### ISSUE TITLE
Edit component empty

### ISSUE THREAD

**Issue by HamAndRock** [marmelab/react-admin#5228]
**What you were expecting:**
To render edit form

**What happened instead:**
Nothing rendered

**Steps to reproduce:**
Create new resource, add Edit component and add if in dataProvider.

Data provider implementation:
![image](https://user-images.githubusercontent.com/24235083/92164649-f6d43080-ee35-11ea-9833-f09d8ebce7c4.png)

Edit form implementation:
![image](https://user-images.githubusercontent.com/24235083/92164703-0e131e00-ee36-11ea-9171-2e4477584e5c.png)

Resource implementation:
![image](https://user-images.githubusercontent.com/24235083/92164755-2420de80-ee36-11ea-95de-2e019ecdde4d.png)

Result:
![image](https://user-images.githubusercontent.com/24235083/92164869-5e8a7b80-ee36-11ea-9ff7-1eeff2cff8d7.png)

**Other information:**
I can get data from dataProvider when I use `useDataProvider().getOne` it works, hovewer when I use just useGetOne it returns undefined again.

Request like create, list and even some other edits work.

**Environment**

* React-admin version: 3.8.2
* React version: 16.13.1
* Browser: Chrome
* Stack trace (in case of a JS error): none

**Comment by djhi** [comment]
Thanks for reporting this. Please provide a sample application showing the issue by forking the following CodeSandbox (https://codesandbox.io/s/github/marmelab/react-admin/tree/master/examples/simple).

**Comment by HamAndRock** [comment]
Sorry for such a late response. I have been digging in the code and problem is with redux. This is how data are represented in redux storage. The key for edited item is not its id but `undefined`.

![image](https://user-images.githubusercontent.com/24235083/93327804-25311300-f81b-11ea-858c-a5b89b710cc8.png)


Other edit components works fine.

![image](https://user-images.githubusercontent.com/24235083/93327958-5e698300-f81b-11ea-8f42-55a2606b5a17.png)

**Comment by MicroJackson** [comment]
I have this exact same behaviour when my API call does not return an ID in his response described here:
https://marmelab.com/react-admin/DataProviders.html#example-implementation
So make sure there is an: 
`"id": 123` or something in your response. I do not see that in your response in the console.
An improvement for react-admin would be to notify the programmer of this fact.

**Comment by HamAndRock** [comment]
Yes, and also the if you supply different id then in url it also doesn't work.
I took me 6 hours to figure out ....

Please can you add some kind of console warning ?

**Comment by fzaninotto** [comment]
There is, indeed, a console warning if the return from `dataProvider.getOne()` doesn't contain an `id` key:

https://github.com/marmelab/react-admin/blob/14c31117c925d832da06c5cd2554d98f63bf91d6/packages/ra-core/src/dataProvider/validateResponseFormat.ts#L43-L51

But it doesn't show up if your `NODE_ENV` variable is set to `production`. Could it be that problem?

**Comment by MicroJackson** [comment]
No, my NODE_ENV is in development mode, and I have never changed it in all 3 of my react-admin projects.
![image](https://user-images.githubusercontent.com/39806531/93641313-5d437c00-f9fc-11ea-9558-21f2f19a2ad8.png)
I tried to make a codesandbox to demonstrate it, but the DataProvider can not give the wrong id, or omit the id.

So I tried to create a simple example:
![image](https://user-images.githubusercontent.com/39806531/93643335-19527600-fa00-11ea-9e61-3d99c4b4b46c.png)
The dataProvider.getOne() fetches that.
![image](https://user-images.githubusercontent.com/39806531/93644965-4e13fc80-fa03-11ea-9bb7-abc2b37c72f5.png)
Nothing shows, and there is no log in the console.
And it does do a valid request to my backend:
![image](https://user-images.githubusercontent.com/39806531/93645130-af3bd000-fa03-11ea-932b-0d7196db0391.png)
So I do not know when that piece of code needs to be triggered, but it is not being triggered for me. Do note that I am using my own dataprovider, same as OP.
